scripts/ci_test_jobs.sh

```diff
@@ -57,13 +57,15 @@ if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then trap "delete_jobs" EXIT; fi
 run_tests
 if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then delete_jobs; fi
 
-# Rebuild in static memory mode.
-cmake .. -DIOT_BUILD_TESTS=1 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS="$CMAKE_FLAGS -DIOT_STATIC_MEMORY_ONLY=1"
-make -j2 aws_iot_tests_jobs
-
-# Run tests in static memory mode.
-if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then create_jobs; fi
-run_tests
-if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then delete_jobs; fi
+if [ "$RUN_TEST" != "coverage" ]; then
+    # Rebuild in static memory mode.
+    cmake .. -DIOT_BUILD_TESTS=1 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS="$CMAKE_FLAGS -DIOT_STATIC_MEMORY_ONLY=1"
+    make -j2 aws_iot_tests_jobs
 
+    # Run tests in static memory mode.
+    if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then create_jobs; fi
```

Comment:
The repeated checking of `$TRAVIS_PULL_REQUEST` is a code smell.  We should put a TODO that this script needs refactoring.

Reply:
Yes, I agree that these scripts should accept configuration flags to alter their behavior.
I suppose I can add a hygiene story for it

Reply:
Added a story that captures all the identifier hygiene needed in these scripts: https://sim.amazon.com/issues/TS-10291